common/dds_specific/srcC/DynamicDataFunc.c

```diff
@@ -0,0 +1,85 @@
+/******************************************************************************/
+/* (c) 2020 Copyright, Real-Time Innovations, Inc. (RTI) All rights reserved. */
+/*                                                                            */
+/* RTI grants Licensee a license to use, modify, compile, and create          */
+/* derivative works of the software solely for use with RTI Connext DDS.      */
+/* Licensee may redistribute copies of the software provided that all such    */
+/* copies are subject to this license.                                        */
+/* The software is provided "as is", with no warranty of any type, including  */
+/* any warranty for fitness for any purpose. RTI is under no obligation to    */
+/* maintain or support the software.  RTI shall not be liable for any         */
+/* incidental or consequential damages arising out of the use or inability to */
+/* use the software.                                                          */
+/*                                                                            */
+/******************************************************************************/
+
+#include <ndds/ndds_c.h>
+
+#include "DynamicDataFunc.h"
+
+DDS_TypeCode * RTI_COMMON_TypeCode_get_member_type(
+    DDS_TypeCode * self,
+    const char * member_name)
+{
+    DDS_UnsignedLong id = 0;
+    DDS_TypeCode *member_tc = NULL;
+    DDS_ExceptionCode_t ex;
+    char *original_name = NULL;
+
+    /* Determine whether it is a nested member or not */
+    if (strchr(member_name, '.') == NULL) {
+        /* If it is not nested */
+        id = DDS_TypeCode_find_member_by_name(self, member_name, &ex);
+        if (ex != DDS_NO_EXCEPTION_CODE) {
+            goto done; /* returns NULL */
+        }
+        member_tc = DDS_TypeCode_member_type(self, id, &ex);
+        if (ex != DDS_NO_EXCEPTION_CODE) {
+            goto done; /* returns NULL */
+        }
+    } else {
+        /* Nested TypeCode */
+        char *token = NULL;
+        const char *delim = ".";
+        char *remaining_string = NULL;
+
+        original_name = DDS_String_dup(member_name);
```

Comment:
Why not just use `member_name` directly? I'm not sure I understand the reason for duplicating it.

Reply:
the function `RTIOsapiUtility_strToken` modifies the content of the input string, it changes the delimitator character to `\0`. For example, if we have `payload.data`, the string will be `payload'\0'data`, and, as the input string is own by the DynamicData, we should create a copy of it.